src/coreclr/inc/check.h

```diff
@@ -684,13 +684,19 @@ CHECK CheckAligned(UINT value, UINT alignment);
 CHECK CheckAligned(ULONG value, UINT alignment);
 #endif
 CHECK CheckAligned(UINT64 value, UINT alignment);
+#ifdef __APPLE__
+CHECK CheckAligned(SIZE_T value, UINT alignment);
+#endif
```

Comment:
Ah, this is something painful that tried to resolve, and get blocked by PAL.

I tried to use template method and `type_traits`, but the common helpers are also included by .c file.

Enumerating the compiler built-in types would be the clearest way that avoids conflict, but the double underscore types are annoying.

Reply:
I agree this is painful. There’s already precedent in the alignment helpers for the same `#if __APPLE__` and `SIZE_T` versions.